src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private Node<K,V>[] table;
+    private int size;
+    private int capacity;
+    private float threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        threshold = DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        size++;
+        int hash = getIndex(key);
+        Node<K, V> newNode = new Node<>(key, value);
+        if (table[hash] == null) {
+            table[hash] = newNode;
+            return;
+        }
+        Node<K, V> currentNode = table[hash];
+        while (currentNode.next != null && !Objects.equals(key,currentNode.key)) {
+            currentNode = currentNode.next;
+        }
+        if (Objects.equals(key, currentNode.key)) {
+            currentNode.value = value;
+            size--;
+            return;
+        }
+        currentNode.next = newNode;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int getIndex(K key) {
+        if (key == null) {
+            return 0;
+        }
+        return Math.abs(key.hashCode()) % capacity;
+    }
+
+    private void resize() {
+        capacity = capacity << 1;
+        threshold = capacity * DEFAULT_LOAD_FACTOR;
+        Node<K, V>[] oldTable = table;
+        table = new Node[capacity];
+        for (Node<K, V> node : oldTable) {
+            while (node != null) {
+                put(node.key, node.value);
+                size--;
```

Comment:
same here - let's remove this line and set size to 0 before transferring nodes into new array